### ISSUE TITLE
[V4] `darkMode` does not work in web

### ISSUE THREAD

**Issue by samtsai** [nativewind/nativewind#1104]
**Describe the bug**

Setting color scheme to `dark` adds `undefined` to body class. 
<img width="351" alt="Screenshot 2024-09-27 at 2 43 43 PM" src="https://github.com/user-attachments/assets/b871a5fa-c449-4e87-8157-864fb37b6a42">

Same issue as #882. There seems to be some sort of timing issue of when the stylesheet is loaded to the DOM and then read from it.

**Reproduction**
See https://github.com/samtsai/my-expo-app

**Expected behavior**
`dark` class should be added and removed to `body`

**Additional context**

I've identified that when [`documentStyle`](https://github.com/nativewind/nativewind/blob/main/packages/react-native-css-interop/src/runtime/web/stylesheet.ts#L4) is set, the styles do not have any CSS variables yet (tailwind, --css-interop, etc.).

This makes the value of [`darkModeValue`](https://github.com/nativewind/nativewind/blob/main/packages/react-native-css-interop/src/runtime/web/color-scheme.ts#L10)  be `undefined`, because the flag has no value.

<img width="1187" alt="Screenshot 2024-09-27 at 2 44 33 PM" src="https://github.com/user-attachments/assets/c8a38009-28d8-4edb-ab0b-55567d242ae9">

**Note** this is only an issue when `app.json` [web output](https://docs.expo.dev/versions/latest/config/app/#output) is set to `single`.

### Workaround

Export the `index.html` via `npx expo export -p web` and manually add the `darkMode` var:
```
:root {
        --css-interop-darkMode: class dark;
      }
```

**Comment by danstepanov** [comment]
Thanks for all the details here @samtsai, I'll look into this with Mark today. @marklawlor is this at all related to react native handling of dark?

**Comment by marklawlor** [comment]
Fixed in `4.1.12`
